torch/signal/windows/windows.py

```diff
@@ -0,0 +1,315 @@
+import torch
+
+from torch import Tensor
+from torch._torch_docs import factory_common_args
+from torch.types import _dtype, _device, _layout
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+
+def _add_docstr(*args):
+    r"""Adds docstrings to a given decorated function.
+
+    Specially useful when then docstrings needs string interpolation, e.g., with
+    str.format().
+    REMARK: Do not use this function if the docstring doesn't need string
+    interpolation, just write a conventional docstring.
+
+    Args:
+        args (str):
+    """
+    def decorator(o):
+        o.__doc__ = ""
+        for arg in args:
+            o.__doc__ += arg
+        return o
+    return decorator
+
+
+def _window_function_checks(function_name: str, window_length: int, dtype: _dtype, layout: _layout) -> None:
+    r"""Performs common checks for all the defined windows.
+     This function should be called before computing any window
+
+     Args:
+         function_name (str): name of the window function.
+         window_length (int): length of the window.
+         dtype (:class:`torch.dtype`): the desired data type of the window tensor.
+         layout (:class:`torch.layout`): the desired layout of the window tensor.
+     """
+
+    def is_floating_type(t: _dtype) -> bool:
+        return t == torch.float32 or t == torch.bfloat16 or t == torch.float64 or t == torch.float16
+
+    def is_complex_type(t: _dtype) -> bool:
+        return t == torch.complex64 or t == torch.complex128 or t == torch.complex32
+
+    if window_length < 0:
+        raise ValueError(f'{function_name} requires non-negative window_length, got window_length={window_length}')
+    if layout is not torch.strided:
+        raise ValueError(f'{function_name} is not implemented for sparse types, got: {layout}')
+    if not is_floating_type(dtype) and not is_complex_type(dtype):
+        raise ValueError(f'{function_name} expects floating point dtypes, got: {dtype}')
+
+
+@_add_docstr(
+    r"""
+Computes a window with an exponential waveform.
+Also known as Poisson window.
+
+The exponential window is defined as follows:
+
+.. math::
+    w(n) = \exp{\left(-\frac{|n - center|}{\tau}\right)}
+    """,
+    r"""
+
+Args:
+    window_length (int): the length of the output window.
+        In other words, the number of points of the ee window.
+    periodic (bool, optional): If `True`, returns a periodic window suitable for use in spectral analysis.
+        If `False`, returns a symmetric window suitable for use in filter design. Default: `True`.
+    center (float, optional): his value defines where the center of the window will be located.
+        In other words, at which sample the peak of the window can be found.
+        Default: `window_length / 2` if `periodic` is `True` (default), else `(window_length - 1) / 2`.
+    tau (float, optional): the decay value.
+        For `center = 0`, it's suggested to use :math:`\tau = -\frac{(M - 1)}{\ln(x)}`,
```

Comment:
Even more, what are `x` and `M`?

Reply:
Well, ok. It seems that the answer is in the "Half-Life" section of the Exponential Decay [here](https://en.wikipedia.org/wiki/Exponential_decay).

Essentially, the half-life is associated with `t1/2` and `x = 2`. In this case, we're interested in a little bit more than half-life but `M-1-life`, right? And `x` would be the "material" left on the right side. Still looking for a better explanation.

Reply:
<img width="974" alt="image" src="https://user-images.githubusercontent.com/13088001/194473082-13a276ec-cc9f-41b5-8705-099d4979bad6.png">

So I guess it can be stated like this: for a given time `M - 1` and the portion of what is left to the right at that time, what the decay value would be? Then, you have `tau = (M-1) / ln(x)`.

Reply:
If it's not clear to us at this moment, it may be better just to remove this for now and let people use whichever values they feel like? We can always update this at a later time

Reply:
Removed. I'll try to come up with a better explanation if there's any.